src/block/Liquid.php

```diff
@@ -500,6 +500,6 @@ protected function liquidCollide(Block $cause, Block $result) : bool{
 	}
 
 	protected function canFlowInto(Block $block) : bool{
-		return $this->pos->getWorld()->isInWorld($block->pos->x, $block->pos->y, $block->pos->z) and $block->canBeFlowedInto() and !($block instanceof Liquid and $block->isSource()); //TODO: I think this should only be liquids of the same type
+		return $this->pos->getWorld()->isInWorld($block->pos->x, $block->pos->y, $block->pos->z) and $block->canBeFlowedInto() and !($block instanceof Liquid and $block->isSource()) and !($block instanceof Air and $block->pos->getWorld()->getChunk($block->pos->x >> 4, $block->pos->z >> 4) === null); //TODO: I think this should only be liquids of the same type
```

Comment:
This check is confusing. It should be refactored to `(!block instanceof Air || isChunkLoaded)`.

(That said, please refer to my earlier comment: this doesn't fix the crash you're referring to).

Reply:
I've refactored as requested. How would this not fix the mentioned crash?